### ISSUE TITLE
Check if pre-defined orders for sub layers work.

### ISSUE THREAD

**Issue by romainmenke** [csstools/postcss-plugins#624]
https://css-tricks.com/css-cascade-layers/#aa-grouping-nested-layers

https://twitter.com/nileshpraj/status/1570828201194029056

```css
@layer one {
  /* sorting the sub-layers */
  @layer two, three;

  /* styles ... */
  @layer three { /* styles ... */ }
  @layer two { /* styles ... */ }
}
```

**Comment by nileshprajapati** [comment]
Thanks for opening the issue,  the following example above works fine.

 However, I was hoping to get the following result if possible.

@layer one {
    /* sorting the sub-layers */
    @layer two, three;
  
    /* styles */
    @import "two" layer(one);
    @import "three" layer(one);
}
    or 

@layer one {
    /* sorting the sub-layers */
    @layer two, three;

    /* styles */
    @layer  two {
       @import "two-one";
        @import "two-two";
    }

    @layer  three {
        @import "three-one";
        @import "three-two";
    }
}

Become:

@layer one {
    /* sorting the sub-layers */
    @layer two, three;
  
    /* styles */
   @layer three { /* styles ... */ }
    @layer two { /* styles ... */ }
}

**Comment by romainmenke** [comment]
`@import` will never work nested in `@layer`.

You need to use this syntax : `@import ... layer(one.two)`

**Comment by romainmenke** [comment]
_(didn't have much time for a longer answer before)_

```css
@layer one {
    /* sorting the sub-layers */
    @layer two, three;
  
    /* styles */
    @import "two" layer(one);
    @import "three" layer(one);
}
```

This is not supported by `postcss-import` or `postcss-cascade-layers` because it is not allowed by the CSS specification.

https://developer.mozilla.org/en-US/docs/Web/CSS/@import#description

It seems that the MDN docs are not entirely up to date with `@layer` but they do mention the important point. `@import` must be declared early and it can not be placed in other rules.

allowed :

```css
@charset ...;

@layer foo, bar;

@import ...;

/* rest of the styles */
```

⚠️ not allowed :

```css
@media screen {
  @import ...; /* not allowed in `@media` */
}

@import ...; /* not allowed after `@media` */

@layer foo {
  @import ...; /* not allowed in `@layer` */
}
```

----------------

That said, there was a real bug with the example from the article.
This update will resolve this issue : https://github.com/csstools/postcss-plugins/pull/625

**Comment by Antonio-Laguna** [comment]
@nileshprajapati this has been released with version `1.1.1` with @romainmenke fix. Thanks again for your report!
